src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRules.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules
+{
+    /// <summary>
+    /// An engine that enforce the execution and validation of consensus rule. 
+    /// </summary>
+    /// <remarks>
+    /// In order for a block to be valid it has to successfully pass the rules checks.
+    /// A block  that is not valid will result in the <see cref="BlockValidationContext.Error"/> as not <c>null</c>.
+    /// </remarks>
+    public interface IConsensusRules
+    {
+        /// <summary>
+        /// Register a new rule to the engine
+        /// </summary>
+        /// <typeparam name="TRule"></typeparam>
+        /// <returns></returns>
+        ConsensusRules Register<TRule>() where TRule : ConsensusRule, new();
```

Comment:
what are you trying to enforce by having `new()`?

Reply:
I will delete this method